src/components/NewMessageForm/NewMessageForm.vue

```diff
@@ -286,25 +262,23 @@ export default {
 		},
 
 		/**
-		 * Uploads the files to the root files directory
+		 * Uploads and shares the selected files
 		 * @param {object} event the file input event object
 		 */
 		async processFiles(event) {
-			// The selected files array
+			// The selected files array coming from the input
 			const files = Object.values(event.target.files)
-			// process each file in the array
-			for (let i = 0; i < files.length; i++) {
-				const userId = this.$store.getters.getUserId()
-				const path = `/files/${userId}/` + files[i].name
-				try {
-					// Upload the file
-					await client.putFileContents(path, files[i])
-					// Share the file to the talk room
-					this.sendFile('/' + files[i].name)
-				} catch (exception) {
-					console.debug('Error while uploading file:' + exception)
-					showError(t('spreed', 'Error while uploading file'))
-				}
+			// Process these files in the store
+			await this.$store.dispatch('addFilesToBeUploaded', { token: this.token, files })
+			// Dispatch the upload action (see uploadStore)
+			await this.$store.dispatch('uploadFiles', this.token)
+			// Get the files that have successfully been uploaded from the store
+			const shareableFiles = this.$store.getters.getShareableFiles(this.token)
+			// Share each of those files in the conversation
+			for (const index in shareableFiles) {
+				// The pat of the file to share to the conversation
+				const path = '/' + shareableFiles[index].name
+				shareFile(path, this.token)
```

Comment:
this-token should not be used after await, due to potential changes
Assign to a variable before the first await